### PR TITLE
Add first half of data ingestion

### PR DESCRIPTION
**Pull request by imbstack**
## Proposed changes

This works afaict. Next I want to add tests and add a script for importing from BQ directly. I want to land this before I do the rest of it to unblock other changes to models.

## Types of changes

What types of changes does your code introduce?
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [guides](https://github.com/mozilla-it/ctms-api/tree/main/guides)
- [ ] I have followed the [Mozilla Lean Data Policies](https://www.mozilla.org/en-US/about/policy/lean-data/)
- [ ] Lint and tests pass both locally and on the cicd with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

_Use this space for additional comments, complications, learnings, alternatives,..._

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -64,6 +64,7 @@ testpaths = [
         "R0912", # Same idea as above, let's leave it since it is a matter of taste
         "W0511", # Let us have our TODOs
         "W0613", # sometimes unused arguments are nice for symmetry and also api methods
+        "R0201", # Not using "self" when it is defined is a matter of taste
```
Comment (bsieber-mozilla) [inline pyproject.toml:67]: 😆

File: ctms/schemas/newsletter.py [hunk 2]
```diff
@@ -43,3 +45,23 @@ class Config:
...
+        description=EMAIL_ID_DESCRIPTION,
+        example=EMAIL_ID_EXAMPLE,
+    )
+    create_timestamp: datetime = Field(
+        default=None,
+        description="Newsletter subscription data creation timestamp",
+        example="2020-12-05T19:21:50.908000+00:00",
+    )
+    update_timestamp: datetime = Field(
+        default=None,
```
Comment (bsieber-mozilla) [inline ctms/schemas/newsletter.py:61]: 'None' defaults for these?

File: ctms/schemas/fxa.py [hunk 3]
```diff
@@ -51,3 +53,21 @@ class Config:
...
+        example=EMAIL_ID_EXAMPLE,
+    )
+    create_timestamp: datetime = Field(
+        description="FXA data creation timestamp",
+        example="2020-12-05T19:21:50.908000+00:00",
+    )
+    update_timestamp: datetime = Field(
+        description="FXA data update timestamp",
+        example="2021-02-04T15:36:57.511000+00:00",
+    )
```
Comment (bsieber-mozilla) [inline ctms/schemas/fxa.py:70]: No defaulting for timestamps here?

File: ctms/schemas/addons.py [hunk 4]
```diff
@@ -46,7 +47,7 @@ class AddOnsBase(ComparableBase):
     )
     location: Optional[str] = Field(
         default=None,
-        max_length=10,
+        max_length=255,
```
Comment (bsieber-mozilla) [inline ctms/schemas/addons.py:50]: I saw the related alembic script for this ✅

File: ctms/models.py [hunk 5]
```diff
@@ -120,7 +120,7 @@ class AmoAccount(Base):
     email_opt_in = Column(Boolean)
     language = Column(String(5))
     last_login = Column(Date)
-    location = Column(String(10))
+    location = Column(String(255))
```
Comment (bsieber-mozilla) [inline ctms/models.py:123]: related alembic migration ✅

**Review by jwhitlock — APPROVED** [review]
Thanks @imbstack, the code looks good! I haven't run it, but I'm assuming you have a few times.

File: ctms/ingest.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+from ctms.schemas import (
+    AddOnsTableSchema,
+    EmailTableSchema,
+    FirefoxAccountsTableSchema,
+    NewsletterTableSchema,
+    VpnWaitlistTableSchema,
+)
+
+
+@dataclass
```
Comment (bsieber-mozilla) [inline ctms/ingest.py:19]: Did you want to use the pydantic version here? https://pydantic-docs.helpmanual.io/usage/dataclasses/
Fine as is as well!

**Review by bsieber-mozilla — APPROVED** [review]
Overall LGTM, some questions if defaults are necessary or not, or maybe they're all expected to have them

File: ctms/schemas/newsletter.py (hunk 2 above)
Comment (imbstack) [reply ctms/schemas/newsletter.py:61]: Oh yeah, we should remove these defaults although I think they do nothing here technically. good catch

File: ctms/ingest.py (hunk 6 above)
Comment (imbstack) [reply ctms/ingest.py:19]: That's true, I figured the minimal thing was good enough for this purpose. Maybe I'll reach for pydantic next time just for consistency

**Pull request merged**